README.md

```diff
@@ -0,0 +1,112 @@
+# Secure AWS Instance Profiles on DC/OS
+Security is always an important topic in today's distributed systems. With DC/OS Enterprise we offer a feature called DC/OS Secrets. With Secrets it is possible to inject secure information like passwords or cryptographic keys into your application. No other Application is able to read or change this information and with DC/OS IAM you can also strip down the group of users having access to this information.
+
+## The usual workflow
+Let's assume you have an application that wants to access AWS resources, like a S3 bucket. With Secrets you can easily create an IAM user and assign it a Policy which is able to access your particular bucket. You take the ACCESS_KEY and add it to your Marathon specification and store the SECRET_ACCESS_KEY into a DC/OS Secret in your default vault which you also specify in your marathon application. This is absolutely not bad but it means you have to rotate these credentials on a regular basis and therefore you need to update your application from time to time.
+
+## AWS Agents can do better
+If your agents are already running on AWS Instances there is a way better and best practise solution solving your problem: Instance Profiles. Instance profiles allow you to assign Roles to Instances. The AWS SDK will realise running on an AWS Instance and try to retrieve credentials from the AWS Metadata API. The huge benefit of this is you do not need to rotate credentials by yourself as AWS is taking care of it. These credentials will have a short live time so even if they get leaked a user will only have a certain amount of time to use those credentials.
```

Comment:
If your agents are already running on AWS Instances there is a way better and best practice solution solving your problem: Instance Profiles. Instance profiles allow you to assign Roles to Instances. The AWS SDK running on an AWS Instance will try to retrieve credentials from the AWS Metadata API. The huge benefit of this is that you do not need to rotate credentials by yourself as AWS is takes care of it. These credentials will have a short lifetime, so even if they get leaked a user will only have a certain amount of time to use those credentials.